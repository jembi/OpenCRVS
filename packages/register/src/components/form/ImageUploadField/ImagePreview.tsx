import * as React from 'react'
import styled from '@register/styledComponents'
import { IFileValue } from '@register/forms'
import { ActionPage } from '@opencrvs/components/lib/interface'
import { OverlayContainer } from '@register/components/form/ImageUploadField/ImageUploadOption'

const PreviewContainer = styled(OverlayContainer)`
  background: ${({ theme }) => theme.colors.black};
`
const ImageHolder = styled.div`
  position: relative;
  width: 100%;
  height: calc(100vh - 90px);
`
const Image = styled.img`
  position: absolute;
  max-width: 100vw;
  max-height: 100%;
  left: 50%;
  top: 50%;
  transform: translateX(-50%) translateY(-50%);
`

type IProps = {
  previewImage: IFileValue
  backLabel?: string
  title?: string
  goBack: () => void
}

export class ImagePreview extends React.Component<IProps> {
  render = () => {
    const { previewImage, title, backLabel, goBack } = this.props
    return (
      <PreviewContainer>
        <ActionPage
          title={title ? title : 'Preview'}
          backLabel={backLabel}
          goBack={goBack}
        >
          <ImageHolder>
            <Image id="preview_image_field" src={previewImage.data} />
          </ImageHolder>
        </ActionPage>
      </PreviewContainer>
    )
  }
}
