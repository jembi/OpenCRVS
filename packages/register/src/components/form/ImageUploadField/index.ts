export * from '@register/components/form/ImageUploadField/ImageUploadField'
