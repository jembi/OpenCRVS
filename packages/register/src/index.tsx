import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { injectGlobal } from '@register/styledComponents'
import { App } from '@register/App'
import registerServiceWorker from '@register/registerServiceWorker'
import { createStore } from '@register/store'
import * as actions from '@register/notification/actions'
import { storage } from '@register/storage'
import * as Sentry from '@sentry/browser'
import * as LogRocket from 'logrocket'
import { SubmissionController } from '@register/SubmissionController'

storage.configStorage('OpenCRVS')

// Injecting global styles for the body tag - used only once
// eslint-disable-line
injectGlobal`
  body {
    margin: 0;
    padding: 0;
    overflow-y:overlay;
  }
`

const { store, history } = createStore()

if (
  window.location.hostname !== 'localhost' &&
  window.location.hostname !== '127.0.0.1'
) {
  // setup error reporting using sentry
  Sentry.init({
    dsn: 'https://8f6ba426b20045f1b91528d5fdc214b5@sentry.io/1401900'
  })

  // setup log rocket to ship log messages and record user errors
  LogRocket.init('hxf1hb/opencrvs')

  // Integrate the two
  Sentry.configureScope(scope => {
    scope.addEventProcessor(async event => {
      if (!event.extra) {
        event.extra = {}
      }
      const sessionUrl = await new Promise(resolve => {
        LogRocket.getSessionURL(url => {
          resolve(url)
        })
      })
      event.extra.sessionURL = sessionUrl
      return event
    })
  })
}

function onNewConentAvailable(waitingSW: ServiceWorker | null) {
  if (waitingSW) {
    const action = actions.showNewContentAvailableNotification(waitingSW)
    store.dispatch(action)
  }
}

function onBackGroundSync() {
  const channel = new BroadcastChannel(
    window.config.BACKGROUND_SYNC_BROADCAST_CHANNEL
  )
  channel.onmessage = e => {
    const syncCount = typeof e.data === 'number' ? e.data : 0
    const action = actions.showBackgroundSyncedNotification(syncCount)
    store.dispatch(action)
  }
}

onBackGroundSync()

ReactDOM.render(
  <App store={store} history={history} />,
  document.getElementById('root')
)

registerServiceWorker(onNewConentAvailable)
new SubmissionController(store).start()
