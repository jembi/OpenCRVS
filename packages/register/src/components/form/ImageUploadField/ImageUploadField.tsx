import * as React from 'react'
import { defineMessages, InjectedIntlProps, injectIntl } from 'react-intl'
import styled from 'styled-components'
import { IconAction } from '@opencrvs/components/lib/buttons'
import { Camera } from '@opencrvs/components/lib/icons'
import { IActionProps } from '@opencrvs/components/lib/buttons/Action'
import { ActionTitle } from '@opencrvs/components/lib/buttons/IconAction'
import { FileItem } from '@opencrvs/components/lib/files'
import { IFormSection, IFileValue } from 'src/forms'
import { ImageUploadOption } from './ImageUploadOption'
import { ImagePreview } from './ImagePreview'

const Container = styled.div`
  width: 100%;
  border-bottom: 1px solid ${({ theme }) => theme.colors.background};
  padding-bottom: 35px;
`
const StyledIcon = styled(Camera)`
  border-radius: 2px;
  box-shadow: 0 0 4px 3px rgba(0, 0, 0, 0.1);
  height: 50px;
  width: 50px;
  background-color: ${({ theme }) => theme.colors.accentLight};
`
const PhotoIconAction = styled(IconAction)`
  outline-style: dashed;
  outline-color: ${({ theme }) => theme.colors.accentLight};
  outline-width: 1px;
  min-height: 90px;
  /* stylelint-disable */
  ${ActionTitle} {
    /* stylelint-enable */
    font-size: 18px;
    margin-left: 75px;
    line-height: 24px;
  }
`
const FileViewer = styled.div`
  margin-top: 15px;
`
const FileViewerLabel = styled.label`
  color: ${({ theme }) => theme.colors.primary};
  font-family: ${({ theme }) => theme.fonts.regularFont};
`
const FileItemContainer = styled.div`
  margin-top: 12px;
`
const messages = defineMessages({
  back: {
    id: 'menu.back',
    defaultMessage: 'Back',
    description: 'Back button in the menu'
  },
  uploadedList: {
    id: 'formFields.imageUpload.uploadedList',
    defaultMessage: 'Uploaded:',
    description: 'label for uploaded list'
  },
  delete: {
    id: 'formFields.imageUpload.delete',
    defaultMessage: 'Delete',
    description: 'label for delete a uploaded item'
  },
  preview: {
    id: 'formFields.imageUpload.preview',
    defaultMessage: 'Preview',
    description: 'label for preview a uploaded item'
  },
  upload: {
    id: 'register.form.upload',
    defaultMessage: 'Upload',
    description: 'title for option view'
  }
})

type IProps = {
  files?: IFileValue[]
  optionSection: IFormSection
  onComplete: (files: IFileValue[]) => void
}
type IFullProps = IActionProps & InjectedIntlProps & IProps

type IFullFileValues = IFileValue & {
  title: string
  description: string
}
/* feels weired may need to change a bit */
function augmentFile(file: IFullFileValues): IFullFileValues {
  if (file.optionValues) {
    file.title = file.optionValues[0].toString()
    if (file.optionValues.length > 1) {
      file.description = file.optionValues[1].toString()
    }
  }
  return file
}
class ImageUploadComponent extends React.Component<
  IFullProps,
  {
    showNestedOptionSection: boolean
    previewImage: IFileValue | null
  }
> {
  constructor(props: IFullProps) {
    super(props)
    this.state = {
      showNestedOptionSection: false,
      previewImage: null
    }
  }

  toggleNestedSection = () => {
    this.setState({
      showNestedOptionSection: !this.state.showNestedOptionSection
    })
  }

  closePreviewSection = () => {
    this.setState({ previewImage: null })
  }

  onDelete = (index: number): void => {
    const files = this.props.files ? this.props.files : []
    files.splice(index, 1)
    this.props.onComplete(files)
  }

  onPreview = (file: IFileValue): void => {
    this.setState({ previewImage: file })
  }

  onComplete = (file: IFileValue) => {
    const files = this.props.files ? this.props.files : []
    this.props.onComplete(files.concat(file))
    this.toggleNestedSection()
  }

  render = () => {
    const { title, optionSection, files, intl } = this.props
    const fileList =
      files &&
      files.map((file: IFullFileValues, index: number) => {
        return (
          <FileItemContainer key={index}>
            <FileItem
              id={`file_item_${index}`}
              file={augmentFile(file)}
              deleteLabel={intl.formatMessage(messages.delete)}
              onDelete={() => this.onDelete(index)}
              previewLabel={intl.formatMessage(messages.preview)}
              onPreview={() => this.onPreview(file)}
            />
          </FileItemContainer>
        )
      })
    return (
      <Container>
        <PhotoIconAction
          type="button"
          icon={() => <StyledIcon />}
          title={title}
          {...this.props}
          onClick={this.toggleNestedSection}
        />

        {fileList && fileList.length > 0 && (
          <FileViewer id="file_list_viewer">
            <FileViewerLabel>
              {intl.formatMessage(messages.uploadedList)}
            </FileViewerLabel>
            {fileList}
          </FileViewer>
        )}

        {this.state.showNestedOptionSection && (
          <ImageUploadOption
            option={optionSection}
            title={intl.formatMessage(messages.upload)}
            backLabel={intl.formatMessage(messages.back)}
            onComplete={this.onComplete}
            toggleNestedSection={this.toggleNestedSection}
          />
        )}

        {this.state.previewImage && (
          <ImagePreview
            previewImage={this.state.previewImage}
            title={intl.formatMessage(messages.preview)}
            backLabel={intl.formatMessage(messages.back)}
            goBack={this.closePreviewSection}
          />
        )}
      </Container>
    )
  }
}
export const ImageUploadField = injectIntl<IFullProps>(ImageUploadComponent)
