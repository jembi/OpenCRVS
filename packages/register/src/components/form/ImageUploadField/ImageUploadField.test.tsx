import * as React from 'react'
import { IntlProvider } from 'react-intl'
import { createShallowRenderedComponent } from '@register/tests/util'
import { ImageUploadField } from '@register/components/form/ImageUploadField/ImageUploadField'
import { documentsSection } from '@register/forms/register/fieldDefinitions/birth/documents-section'
import { IImageUploaderWithOptionsFormField } from '@opencrvs/register/src/forms'

describe('image upload component', () => {
  const mock = jest.fn()
  const testComponent = createShallowRenderedComponent(
    <IntlProvider locale="en">
      <ImageUploadField
        id="image_uploader"
        title="Upload a photo of the supporting document"
        onComplete={mock}
        optionSection={
          (documentsSection.fields[0] as IImageUploaderWithOptionsFormField)
            .optionSection
        }
      />
    </IntlProvider>
  )

  it('renders without crashing', () => {
    expect(testComponent).toMatchSnapshot()
  })
})
