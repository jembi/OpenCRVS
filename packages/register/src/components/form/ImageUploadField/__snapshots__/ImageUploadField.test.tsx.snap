// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image upload component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider
    locale="en"
  >
    <InjectIntl(ImageUploadComponent)
      id="image_uploader"
      onComplete={[MockFunction]}
      optionSection={
        Object {
          "fields": Array [
            Object {
              "hideAsterisk": true,
              "initialValue": "",
              "label": Object {
                "defaultMessage": "Whose suppoting document are you uploading?",
                "description": "Question to ask user, for whom, documents are being uploaded",
                "id": "formFields.uploadDocForWhom",
              },
              "name": "uploadDocForWhom",
              "options": Array [
                Object {
                  "label": Object {
                    "defaultMessage": "Child",
                    "description": "Label for radio option Child",
                    "id": "formFields.uploadDocForChild",
                  },
                  "value": "Child",
                },
                Object {
                  "label": Object {
                    "defaultMessage": "Mother",
                    "description": "Label for radio option Mother",
                    "id": "formFields.uploadDocForMother",
                  },
                  "value": "Mother",
                },
                Object {
                  "label": Object {
                    "defaultMessage": "Father",
                    "description": "Label for radio option Father",
                    "id": "formFields.uploadDocForFather",
                  },
                  "value": "Father",
                },
                Object {
                  "label": Object {
                    "defaultMessage": "Other",
                    "description": "Label for radio option Other",
                    "id": "formFields.uploadDocForOther",
                  },
                  "value": "Other",
                },
              ],
              "required": true,
              "type": "RADIO_GROUP",
              "validate": Array [],
            },
            Object {
              "dynamicOptions": Object {
                "dependency": "uploadDocForWhom",
                "options": Object {
                  "Child": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Proof of Place and Date of Birth",
                        "description": "Label for select option Child Birth Proof",
                        "id": "formFields.docTypeChildBirthProof",
                      },
                      "value": "Proof of Place and Date of Birth",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "EPI Card",
                        "description": "Label for select option EPI Card",
                        "id": "formFields.docTypeEPICard",
                      },
                      "value": "EPI Card",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Doctor Certificate",
                        "description": "Label for select option Doctor Certificate",
                        "id": "formFields.docTypeDoctorCertificate",
                      },
                      "value": "Doctor Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                  ],
                  "Father": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Birth Registration",
                        "description": "Label for radio option Birth Registration",
                        "id": "formFields.docTypeBR",
                      },
                      "value": "Birth Registration",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (front)",
                        "description": "Label for select option radio option NID front",
                        "id": "formFields.docTypeNIDFront",
                      },
                      "value": "National ID (front)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (back)",
                        "description": "Label for select option radio option NID back",
                        "id": "formFields.docTypeNIDBack",
                      },
                      "value": "National ID (back)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Passport",
                        "description": "Label for radio option Passport",
                        "id": "formFields.docTypePassport",
                      },
                      "value": "Passport",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Other",
                        "description": "Label for radio option Other",
                        "id": "formFields.docTypeOther",
                      },
                      "value": "Other",
                    },
                  ],
                  "Mother": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Birth Registration",
                        "description": "Label for radio option Birth Registration",
                        "id": "formFields.docTypeBR",
                      },
                      "value": "Birth Registration",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (front)",
                        "description": "Label for select option radio option NID front",
                        "id": "formFields.docTypeNIDFront",
                      },
                      "value": "National ID (front)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (back)",
                        "description": "Label for select option radio option NID back",
                        "id": "formFields.docTypeNIDBack",
                      },
                      "value": "National ID (back)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Passport",
                        "description": "Label for radio option Passport",
                        "id": "formFields.docTypePassport",
                      },
                      "value": "Passport",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Other",
                        "description": "Label for radio option Other",
                        "id": "formFields.docTypeOther",
                      },
                      "value": "Other",
                    },
                  ],
                  "Other": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Birth Registration",
                        "description": "Label for radio option Birth Registration",
                        "id": "formFields.docTypeBR",
                      },
                      "value": "Birth Registration",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (front)",
                        "description": "Label for select option radio option NID front",
                        "id": "formFields.docTypeNIDFront",
                      },
                      "value": "National ID (front)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (back)",
                        "description": "Label for select option radio option NID back",
                        "id": "formFields.docTypeNIDBack",
                      },
                      "value": "National ID (back)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Passport",
                        "description": "Label for radio option Passport",
                        "id": "formFields.docTypePassport",
                      },
                      "value": "Passport",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Other",
                        "description": "Label for radio option Other",
                        "id": "formFields.docTypeOther",
                      },
                      "value": "Other",
                    },
                  ],
                },
              },
              "hideAsterisk": true,
              "initialValue": "",
              "label": Object {
                "defaultMessage": "Which document type are you uploading?",
                "description": "Question to ask user, what type of documents are being uploaded",
                "id": "formFields.whatDocToUpload",
              },
              "name": "whatDocToUpload",
              "required": true,
              "type": "SELECT_WITH_DYNAMIC_OPTIONS",
              "validate": Array [],
            },
          ],
          "id": "documents-upload",
          "name": Object {
            "defaultMessage": "Documents Upload",
            "description": "Tab title for Documents Upload",
            "id": "register.form.section.upload.documentsName",
          },
          "title": Object {
            "defaultMessage": "Supporting documents",
            "description": "Form section title for Documents",
            "id": "register.form.section.upload.documentsTitle",
          },
          "viewType": "form",
        }
      }
      title="Upload a photo of the supporting document"
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "id": "image_uploader",
      "onComplete": [MockFunction],
      "optionSection": Object {
        "fields": Array [
          Object {
            "hideAsterisk": true,
            "initialValue": "",
            "label": Object {
              "defaultMessage": "Whose suppoting document are you uploading?",
              "description": "Question to ask user, for whom, documents are being uploaded",
              "id": "formFields.uploadDocForWhom",
            },
            "name": "uploadDocForWhom",
            "options": Array [
              Object {
                "label": Object {
                  "defaultMessage": "Child",
                  "description": "Label for radio option Child",
                  "id": "formFields.uploadDocForChild",
                },
                "value": "Child",
              },
              Object {
                "label": Object {
                  "defaultMessage": "Mother",
                  "description": "Label for radio option Mother",
                  "id": "formFields.uploadDocForMother",
                },
                "value": "Mother",
              },
              Object {
                "label": Object {
                  "defaultMessage": "Father",
                  "description": "Label for radio option Father",
                  "id": "formFields.uploadDocForFather",
                },
                "value": "Father",
              },
              Object {
                "label": Object {
                  "defaultMessage": "Other",
                  "description": "Label for radio option Other",
                  "id": "formFields.uploadDocForOther",
                },
                "value": "Other",
              },
            ],
            "required": true,
            "type": "RADIO_GROUP",
            "validate": Array [],
          },
          Object {
            "dynamicOptions": Object {
              "dependency": "uploadDocForWhom",
              "options": Object {
                "Child": Array [
                  Object {
                    "label": Object {
                      "defaultMessage": "Proof of Place and Date of Birth",
                      "description": "Label for select option Child Birth Proof",
                      "id": "formFields.docTypeChildBirthProof",
                    },
                    "value": "Proof of Place and Date of Birth",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "EPI Card",
                      "description": "Label for select option EPI Card",
                      "id": "formFields.docTypeEPICard",
                    },
                    "value": "EPI Card",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "Doctor Certificate",
                      "description": "Label for select option Doctor Certificate",
                      "id": "formFields.docTypeDoctorCertificate",
                    },
                    "value": "Doctor Certificate",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "School Certificate",
                      "description": "Label for radio option School Certificate",
                      "id": "formFields.docTypeSC",
                    },
                    "value": "School Certificate",
                  },
                ],
                "Father": Array [
                  Object {
                    "label": Object {
                      "defaultMessage": "Birth Registration",
                      "description": "Label for radio option Birth Registration",
                      "id": "formFields.docTypeBR",
                    },
                    "value": "Birth Registration",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "National ID (front)",
                      "description": "Label for select option radio option NID front",
                      "id": "formFields.docTypeNIDFront",
                    },
                    "value": "National ID (front)",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "National ID (back)",
                      "description": "Label for select option radio option NID back",
                      "id": "formFields.docTypeNIDBack",
                    },
                    "value": "National ID (back)",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "Passport",
                      "description": "Label for radio option Passport",
                      "id": "formFields.docTypePassport",
                    },
                    "value": "Passport",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "School Certificate",
                      "description": "Label for radio option School Certificate",
                      "id": "formFields.docTypeSC",
                    },
                    "value": "School Certificate",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "Other",
                      "description": "Label for radio option Other",
                      "id": "formFields.docTypeOther",
                    },
                    "value": "Other",
                  },
                ],
                "Mother": Array [
                  Object {
                    "label": Object {
                      "defaultMessage": "Birth Registration",
                      "description": "Label for radio option Birth Registration",
                      "id": "formFields.docTypeBR",
                    },
                    "value": "Birth Registration",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "National ID (front)",
                      "description": "Label for select option radio option NID front",
                      "id": "formFields.docTypeNIDFront",
                    },
                    "value": "National ID (front)",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "National ID (back)",
                      "description": "Label for select option radio option NID back",
                      "id": "formFields.docTypeNIDBack",
                    },
                    "value": "National ID (back)",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "Passport",
                      "description": "Label for radio option Passport",
                      "id": "formFields.docTypePassport",
                    },
                    "value": "Passport",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "School Certificate",
                      "description": "Label for radio option School Certificate",
                      "id": "formFields.docTypeSC",
                    },
                    "value": "School Certificate",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "Other",
                      "description": "Label for radio option Other",
                      "id": "formFields.docTypeOther",
                    },
                    "value": "Other",
                  },
                ],
                "Other": Array [
                  Object {
                    "label": Object {
                      "defaultMessage": "Birth Registration",
                      "description": "Label for radio option Birth Registration",
                      "id": "formFields.docTypeBR",
                    },
                    "value": "Birth Registration",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "National ID (front)",
                      "description": "Label for select option radio option NID front",
                      "id": "formFields.docTypeNIDFront",
                    },
                    "value": "National ID (front)",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "National ID (back)",
                      "description": "Label for select option radio option NID back",
                      "id": "formFields.docTypeNIDBack",
                    },
                    "value": "National ID (back)",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "Passport",
                      "description": "Label for radio option Passport",
                      "id": "formFields.docTypePassport",
                    },
                    "value": "Passport",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "School Certificate",
                      "description": "Label for radio option School Certificate",
                      "id": "formFields.docTypeSC",
                    },
                    "value": "School Certificate",
                  },
                  Object {
                    "label": Object {
                      "defaultMessage": "Other",
                      "description": "Label for radio option Other",
                      "id": "formFields.docTypeOther",
                    },
                    "value": "Other",
                  },
                ],
              },
            },
            "hideAsterisk": true,
            "initialValue": "",
            "label": Object {
              "defaultMessage": "Which document type are you uploading?",
              "description": "Question to ask user, what type of documents are being uploaded",
              "id": "formFields.whatDocToUpload",
            },
            "name": "whatDocToUpload",
            "required": true,
            "type": "SELECT_WITH_DYNAMIC_OPTIONS",
            "validate": Array [],
          },
        ],
        "id": "documents-upload",
        "name": Object {
          "defaultMessage": "Documents Upload",
          "description": "Tab title for Documents Upload",
          "id": "register.form.section.upload.documentsName",
        },
        "title": Object {
          "defaultMessage": "Supporting documents",
          "description": "Form section title for Documents",
          "id": "register.form.section.upload.documentsTitle",
        },
        "viewType": "form",
      },
      "title": "Upload a photo of the supporting document",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "id": "image_uploader",
        "onComplete": [MockFunction],
        "optionSection": Object {
          "fields": Array [
            Object {
              "hideAsterisk": true,
              "initialValue": "",
              "label": Object {
                "defaultMessage": "Whose suppoting document are you uploading?",
                "description": "Question to ask user, for whom, documents are being uploaded",
                "id": "formFields.uploadDocForWhom",
              },
              "name": "uploadDocForWhom",
              "options": Array [
                Object {
                  "label": Object {
                    "defaultMessage": "Child",
                    "description": "Label for radio option Child",
                    "id": "formFields.uploadDocForChild",
                  },
                  "value": "Child",
                },
                Object {
                  "label": Object {
                    "defaultMessage": "Mother",
                    "description": "Label for radio option Mother",
                    "id": "formFields.uploadDocForMother",
                  },
                  "value": "Mother",
                },
                Object {
                  "label": Object {
                    "defaultMessage": "Father",
                    "description": "Label for radio option Father",
                    "id": "formFields.uploadDocForFather",
                  },
                  "value": "Father",
                },
                Object {
                  "label": Object {
                    "defaultMessage": "Other",
                    "description": "Label for radio option Other",
                    "id": "formFields.uploadDocForOther",
                  },
                  "value": "Other",
                },
              ],
              "required": true,
              "type": "RADIO_GROUP",
              "validate": Array [],
            },
            Object {
              "dynamicOptions": Object {
                "dependency": "uploadDocForWhom",
                "options": Object {
                  "Child": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Proof of Place and Date of Birth",
                        "description": "Label for select option Child Birth Proof",
                        "id": "formFields.docTypeChildBirthProof",
                      },
                      "value": "Proof of Place and Date of Birth",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "EPI Card",
                        "description": "Label for select option EPI Card",
                        "id": "formFields.docTypeEPICard",
                      },
                      "value": "EPI Card",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Doctor Certificate",
                        "description": "Label for select option Doctor Certificate",
                        "id": "formFields.docTypeDoctorCertificate",
                      },
                      "value": "Doctor Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                  ],
                  "Father": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Birth Registration",
                        "description": "Label for radio option Birth Registration",
                        "id": "formFields.docTypeBR",
                      },
                      "value": "Birth Registration",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (front)",
                        "description": "Label for select option radio option NID front",
                        "id": "formFields.docTypeNIDFront",
                      },
                      "value": "National ID (front)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (back)",
                        "description": "Label for select option radio option NID back",
                        "id": "formFields.docTypeNIDBack",
                      },
                      "value": "National ID (back)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Passport",
                        "description": "Label for radio option Passport",
                        "id": "formFields.docTypePassport",
                      },
                      "value": "Passport",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Other",
                        "description": "Label for radio option Other",
                        "id": "formFields.docTypeOther",
                      },
                      "value": "Other",
                    },
                  ],
                  "Mother": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Birth Registration",
                        "description": "Label for radio option Birth Registration",
                        "id": "formFields.docTypeBR",
                      },
                      "value": "Birth Registration",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (front)",
                        "description": "Label for select option radio option NID front",
                        "id": "formFields.docTypeNIDFront",
                      },
                      "value": "National ID (front)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (back)",
                        "description": "Label for select option radio option NID back",
                        "id": "formFields.docTypeNIDBack",
                      },
                      "value": "National ID (back)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Passport",
                        "description": "Label for radio option Passport",
                        "id": "formFields.docTypePassport",
                      },
                      "value": "Passport",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Other",
                        "description": "Label for radio option Other",
                        "id": "formFields.docTypeOther",
                      },
                      "value": "Other",
                    },
                  ],
                  "Other": Array [
                    Object {
                      "label": Object {
                        "defaultMessage": "Birth Registration",
                        "description": "Label for radio option Birth Registration",
                        "id": "formFields.docTypeBR",
                      },
                      "value": "Birth Registration",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (front)",
                        "description": "Label for select option radio option NID front",
                        "id": "formFields.docTypeNIDFront",
                      },
                      "value": "National ID (front)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "National ID (back)",
                        "description": "Label for select option radio option NID back",
                        "id": "formFields.docTypeNIDBack",
                      },
                      "value": "National ID (back)",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Passport",
                        "description": "Label for radio option Passport",
                        "id": "formFields.docTypePassport",
                      },
                      "value": "Passport",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "School Certificate",
                        "description": "Label for radio option School Certificate",
                        "id": "formFields.docTypeSC",
                      },
                      "value": "School Certificate",
                    },
                    Object {
                      "label": Object {
                        "defaultMessage": "Other",
                        "description": "Label for radio option Other",
                        "id": "formFields.docTypeOther",
                      },
                      "value": "Other",
                    },
                  ],
                },
              },
              "hideAsterisk": true,
              "initialValue": "",
              "label": Object {
                "defaultMessage": "Which document type are you uploading?",
                "description": "Question to ask user, what type of documents are being uploaded",
                "id": "formFields.whatDocToUpload",
              },
              "name": "whatDocToUpload",
              "required": true,
              "type": "SELECT_WITH_DYNAMIC_OPTIONS",
              "validate": Array [],
            },
          ],
          "id": "documents-upload",
          "name": Object {
            "defaultMessage": "Documents Upload",
            "description": "Tab title for Documents Upload",
            "id": "register.form.section.upload.documentsName",
          },
          "title": Object {
            "defaultMessage": "Supporting documents",
            "description": "Form section title for Documents",
            "id": "register.form.section.upload.documentsTitle",
          },
          "viewType": "form",
        },
        "title": "Upload a photo of the supporting document",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
